TransIT.DAL/Repositories/BaseRepository.cs

```diff
@@ -17,53 +17,54 @@ public abstract class BaseRepository<TEntity> : IBaseRepository<TEntity>, IQuery
 
         public BaseRepository(TransITDBContext context)
         {
+            _entities = context.Set<TEntity>();
             _context = context;
         }
 
         public virtual Task<TEntity> GetByIdAsync(int id)
         {
-            return ComplexEntities.SingleOrDefaultAsync(t => t.Id.Equals(id));
+            return ComplexEntities.Where(t => t.Id == id).FirstOrDefaultAsync();
         }
 
-        public virtual Task<IEnumerable<TEntity>> GetAllAsync()
+        public async virtual Task<IEnumerable<TEntity>> GetAllAsync()
```

Comment:
remove async await keywords from all methods that dont need it(one-line methods)